archunit/src/test/java/com/tngtech/archunit/lang/syntax/elements/MembersShouldTest.java

```diff
@@ -672,6 +680,26 @@ public void containNumberOfElements_fails_on_mismatching_predicate() {
                         + "com.tngtech.archunit.lang.syntax.elements.testclasses.SimpleFieldAndMethod.violated()]");
     }
 
+    @Test
+    public void should_fail_on_empty_should_by_default() {
+        assertThatThrownBy(new ThrowableAssert.ThrowingCallable() {
+            @Override
+            public void call() {
+                members().that().haveName("doesNotExist").should().bePublic()
```

Comment:
We could extract a semantic constant
```java
private ArchRule ruleWithEmptyShould = members().that().haveName("doesNotExist").should().bePublic();
```

Reply:
The problem about extracting this further as directly locally above the line where it is used is that it needs the combination of the imported classes and the rule. The rule by itself at the moment can't be `emptyShould` per se, only if you take into consideration that we're importing a specific class.
But thinking about it we can just use `alwaysFalse` as predicate, then it's alright again :+1: